src/policies/security.ts

```diff
@@ -0,0 +1,233 @@
+
+    // Check Pod level security context
+    const podProfileType = request.Raw.spec?.securityContext?.seccompProfile?.type || "";
+    if (!authorizedTyes.includes(podProfileType)) {
+      return request.Deny(`The seccomp profile at Pod level is not in the allowed list.`);
+    }
+
+    const hasOtherSecProfileType = containers(request)
+      .flatMap(c => c.securityContext?.seccompProfile)
+      .find(profile => !authorizedTyes.includes(profile?.type || ""));
+
+    if (hasOtherSecProfileType) {
+      return request.Deny(`Unauthorized seccomp profile type.`);
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * Restrict SELinux Type in Pods
+ *
+ * SELinux options can be used to escalate privileges. This policy ensures that the
+ * `seLinuxOptions` type field is set to either `container_t`, `container_init_t`, or
+ * `container_kvm_t`. Applies to Pods and all types of containers within them.
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-selinux-type.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    if (exemptSELinuxTypes(request)) {
+      return request.Approve();
+    }
+
+    const allowedSeLinuxTypes = ["container_t", "container_init_t", "container_kvm_t"];
+
+    // Check Pod level security context
+    const podSeLinuxType = request.Raw.spec?.securityContext?.seLinuxOptions?.type || "";
+    if (!allowedSeLinuxTypes.includes(podSeLinuxType)) {
+      return request.Deny(`Setting SELinux type at Pod level is restricted to the allowed list.`);
+    }
+
+    const hasOtherSELinuxType = containers(request)
+      .flatMap(c => c.securityContext?.seLinuxOptions?.type)
+      .find(t => !allowedSeLinuxTypes.includes(t || ""));
+
+    if (hasOtherSELinuxType) {
+      return request.Deny(`Unauthorized SELinux type.`);
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * Drop All Capabilities in Pods
+ *
+ * This policy ensures that all containers, initContainers, and ephemeralContainers in a Pod
+ * explicitly specify `drop: ["ALL"]` in their securityContext capabilities. Capabilities permit
+ * privileged actions without giving full root access. Dropping all capabilities and only adding
+ * back those that are required increases the security posture of the Pod.
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/require-drop-all-capabilities.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    if (exemptDropAllCapabilities(request)) {
+      return request.Approve();
+    }
+
+    const hasInvalidCapabilities = containers(request)
+      .flatMap(c => c.securityContext?.capabilities?.drop || [])
+      .find(c => c !== "ALL");
+
+    if (hasInvalidCapabilities) {
+      return request.Deny(
+        `Containers must drop all Linux capabilities by setting spec.containers[*].securityContext.capabilities.drop to 'ALL'.`,
```

Comment:
nit: this could also be init/ephemeral containers which the deny message doesn't capture.